engine-runner-bin/src/main.rs

```diff
@@ -77,3 +80,100 @@ fn main() -> anyhow::Result<()> {
 	}
 	Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+
+	use engine_upgrade_utils::ERROR_READING_SETTINGS;
+
+	use super::*;
+
+	// This tests:
+	// 1. We can run by providing all necessary settings via command line.
+	// 2. That args are filtered out correctly before passing them into the old version.
+	// 3. That the old version can run with the filtered args. implying that the settings in the new
+	//    version are backwards compatible.
+	#[test]
+	fn incompatible_args_should_be_filtered_out() {
+		// create temporary file
+		let tempdir = tempfile::TempDir::new().unwrap();
```

Comment:
Does `tempfile` automatically clean up the temporary file that is created? Maybe we should delete it?

Reply:
Should be cleaned up on Drop: https://docs.rs/tempfile/latest/tempfile/#resource-leaking